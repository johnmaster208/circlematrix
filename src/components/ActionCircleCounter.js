import React from 'react';
import PropTypes from 'prop-types';


const ActionCircleCounter = (props) => {
  const { counter } = props

  return(
    <div className="text-center">
      <div className="circle-counter-number">{counter}</div>
      <div className="circle-counter-cr">circles remaining</div>
    </div>
  );


};

ActionCircleCounter.propTypes = {
  counter: PropTypes.number
}

export default ActionCircleCounter
