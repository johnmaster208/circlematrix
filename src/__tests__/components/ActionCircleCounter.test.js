import React from 'react';
import ActionCircleCounter from '../../components/ActionCircleCounter';
import { shallow } from 'enzyme';

describe('ActionCircleCounter component', () => {

    const props = {
        counter: 7
    };

    it('Renders without crashing...', () => {
        const wrapper = shallow(
            <ActionCircleCounter {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});