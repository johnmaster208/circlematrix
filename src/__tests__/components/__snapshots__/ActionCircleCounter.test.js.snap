// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCircleCounter component Renders without crashing... 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionCircleCounter
    counter={7}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="circle-counter-number"
>
          7
</div>,
        <div
          className="circle-counter-cr"
>
          circles remaining
</div>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 7,
          "className": "circle-counter-number",
        },
        "ref": null,
        "rendered": 7,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "circles remaining",
          "className": "circle-counter-cr",
        },
        "ref": null,
        "rendered": "circles remaining",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="circle-counter-number"
>
            7
</div>,
          <div
            className="circle-counter-cr"
>
            circles remaining
</div>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "className": "circle-counter-number",
          },
          "ref": null,
          "rendered": 7,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "circles remaining",
            "className": "circle-counter-cr",
          },
          "ref": null,
          "rendered": "circles remaining",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
